src/test/java/de/medizininformatikinitiative/torch/rest/FhirControllerWebTestClientTest.java

```diff
@@ -0,0 +1,286 @@
+                    .exchange()
+                    .expectStatus().isOk()
+                    .expectHeader().contentType(MediaType.APPLICATION_JSON)
+                    .expectBody()
+                    .jsonPath("$.job1").isEqualTo("202 Accepted")
+                    .jsonPath("$.job2").isEqualTo("200 OK");
+        }
+    }
+
+    @Test
+    void checkAcceptedStatus() {
+        String jobId = "accepted-job";
+
+        when(resultFileManager.getStatus(jobId)).thenReturn(HttpStatus.ACCEPTED);
+
+        webTestClient.get()
+                .uri("/fhir/__status/{jobId}", jobId)
+                .exchange()
+                .expectStatus().isEqualTo(HttpStatus.ACCEPTED)
+                .expectBody().isEmpty();
+    }
+
+    @ParameterizedTest
+    @CsvSource({
+            "ok,       application/json,        200",
+            "ok-null,  text/plain,              404",
+            "error,    application/fhir+json,   500",
+            "read-fail,text/plain,              404"
+    })
+    void loadStatusBranches(String scenario, String expectedContentType, int expectedStatus) {
+        String jobId = "test-job";
+
+        switch (scenario) {
+            case "ok" -> {
+                when(resultFileManager.getStatus(jobId)).thenReturn(HttpStatus.OK);
+                when(resultFileManager.loadBundleFromFileSystem(eq(jobId), anyString()))
+                        .thenReturn(Map.of("message", "bundle content"));
+            }
+            case "ok-null" -> {
+                when(resultFileManager.getStatus(jobId)).thenReturn(HttpStatus.OK);
+                when(resultFileManager.loadBundleFromFileSystem(eq(jobId), anyString()))
+                        .thenReturn(Map.of());
+            }
+            case "accepted" -> when(resultFileManager.getStatus(jobId))
+                    .thenReturn(HttpStatus.ACCEPTED);
+            case "error" -> {
+                when(resultFileManager.getStatus(jobId)).thenReturn(HttpStatus.INTERNAL_SERVER_ERROR);
+                when(resultFileManager.loadErrorFromFileSystem(jobId))
+                        .thenReturn("{\"resourceType\":\"OperationOutcome\"}");
+            }
+            case "read-fail" -> {
+                when(resultFileManager.getStatus(jobId)).thenReturn(HttpStatus.INTERNAL_SERVER_ERROR);
+                when(resultFileManager.loadErrorFromFileSystem(jobId))
+                        .thenAnswer(invocation -> {
+                            throw new IOException("disk failure");
+                        });
+            }
+        }
+
+        Mono<String> body = webTestClient.get()
+                .uri("/fhir/__status/{jobId}", jobId)
+                .exchange()
+                .expectStatus().isEqualTo(expectedStatus)
+                .expectHeader().contentType(expectedContentType)
+                .returnResult(String.class)
+                .getResponseBody()
+                .single();
+
+        StepVerifier.create(body)
```

Comment:
sorry, I was not aware that the client has already assertions built-in. So you can also use them